plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py

```diff
@@ -0,0 +1,70 @@
+import typing
+
+import polars as pl
+
+from flytekit import FlyteContext
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.models.types import StructuredDatasetType
+from flytekit.types.structured.structured_dataset import (
+    PARQUET,
+    LOCAL,
+    S3,
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetTransformerEngine,
+)
+
+
+class PolarsDataFrameToParquetEncodingHandler(StructuredDatasetEncoder):
+    def __init__(self, protocol: str):
+        super().__init__(pl.DataFrame, protocol, PARQUET)
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        df = typing.cast(pl.DataFrame, structured_dataset.dataframe)
+
+        local_dir = typing.cast(str, structured_dataset.uri) or ctx.file_access.get_random_local_directory()
```

Comment:
I'm not sure about this part. In the numpy example [here](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/type_system/structured_dataset.html) it only has `ctx.file_access.get_random_local_directory()` but the Spark plugin had this extra `typing.cast` part.